gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -484,7 +484,9 @@ public synchronized void setActive(boolean active) {
           log.error("Exception encountered when shutting down DagManager threads.", e);
         }
       }
-    } catch (IOException e) {
+    } catch (Throwable e) {
```

Comment:
1) With the current code (i.e. the left side) also, leader transition should fail. I did not get what are we trying to achieve here.
2) Is GobblinServiceManager (the caller of this method) not printing stacktrace?

Reply:
We are not catching `NullPointerException` that occurred here https://github.com/apache/gobblin/blob/9e30c6c3bca857942fba5d10345b1be14adc1942/gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java#L435 we're only catching IOException which results in leader transition silently not completing if NPE or something occurs. It is printing the stack trace but we already set `this.active = true beforehand` so it remains as leader.

Reply:
I agree w/ the spirit here, but `Throwable` is pretty severe to catch... it would even grab OOM and other `Error`s... do we want that, or should you merely catch `Exception`?

Reply:
1) why do we set this.active=true when it is not ready yet to act as a leader?
2) should we just reverse the leadership change in the catch block then?
3) basically the only difference this change is making is printing a log line. lets try to handle the exception(including runtime) gracefully.

Reply:
I am updating it to catch `RuntimeException` then. How about that for the error catching? 
I was thinking about setting this.active after we finish all the work, but the reason we may set it before is that loading all dags from state store and initializing may take time but in meantime we still want to accept new requests. I want to limit scope of significant change like that which may increase downtime.

Reply:
Catching and throwing the exception will fall the service health check so the service deploy/startup will fail if we encounter NPE. Previously it would silently fail initialization is the issue.